### DISCUSSION TITLE
flux will not recover from failed deployments

### DISCUSSION THREAD

**Discussion by alita-moore** [fluxcd/flux2#5115] [Q&A]
I might be doing something wrong but when a deployment fails (i.e. there is an ImagePullBackoff error or  a config error) flux fails to re-deploy the deployment so I have to manually run `kubectl rollout deployment whatever` for the reconciliation to get unstuck. how can I make flux do this automatically?

**Comment by stefanprodan** [comment]
Flux will redeploy it after the timeout expires, you should also set the retry interval, see here the recommended settings: https://fluxcd.io/flux/components/kustomize/kustomizations/#recommended-settings

**Reply by alita-moore** [reply]
it seems that even with these settings it's not recovering / retrying deployments for knative services

**Reply by alita-moore** [reply]
i.e. it would be potentially a good idea to force a new revision of the knative service in the event of a timeout / failed deployment. Doing so manually resolves the problem. (I do so by setting some annotation on the knative service).
